### PR TITLE
Display reactions for issue's comments

### PR DESCRIPTION
**Pull request by vkavalchuk**
before|after
---|---
before|after

### CODE REVIEW

**Comment by jonan** [comment]
Thanks for the PR, really nice feature. Anyhow, there's a simpler way of solving this issue.

The comments are already fetched using the Timeline API, and simply adding the header `"Accept: application/vnd.github.squirrel-girl-preview"` to that requests is enough to get the reactions. So you don't need to fetch all comments again, and it'll work fine with pagination (your approach wont work for issues with >100 comments).

Unfortunately, I tried this and it doesn't work due to a bug in the API (the `body_html` param, which we use to show the comment's text, is not returned), so we can't implement it just yet. I've reported it to GitHub and hope they can fix it soon so we can have this feature in the next release.

You can try to fix this PR and I will merge it once GitHub fixes the bug.

**Comment by vkavalchuk** [comment]
> The comments are already fetched using the Timeline API, and simply adding the header "Accept: application/vnd.github.squirrel-girl-preview" to that requests is enough to get the reactions.

Unfortunately, [timeline request](https://developer.github.com/v3/issues/timeline/) accepts another header `"Accept: application/vnd.github.mockingbird-preview"`. I've got an error when tried `"Accept: application/vnd.github.squirrel-girl-preview"`. I tried to use [reaction request](https://developer.github.com/v3/reactions/#list-reactions-for-an-issue-comment), but I got too many information - list of reactions including user info wich needed to be agregate (List\<Reaction\> to ReactionSummary). Finally I choosed this way to get all reactions by one request.
Founded one trouble while developing. Timeline refreshes for three or more different issues when user clicks on only one issue to view details. For example click on issue 300 fires 299, 300, 301 issue details updating. IMHO that's a real reason of slow refreshing repository's issue info.

> So you don't need to fetch all comments again, and it'll work fine with pagination (your approach wont work for issues with >100 comments).

So pagination service not used properly:(
`Collection<TimelineEvent> timelineEvents = paginationService.getAll();`
Yep, there is a perfomance problem with big amount of comments. GraphQL API v4 may partially resolve a problem, but don't know how to use it yet.

**Comment by jonan** [comment]
> Unfortunately, [timeline request](https://developer.github.com/v3/issues/timeline/) accepts another header `"Accept: application/vnd.github.mockingbird-preview"`. I've got an error when tried `"Accept: application/vnd.github.squirrel-girl-preview"` even with separated getTimeline1() method.

You can add various headers like this:
```
@Headers({
    "Accept: application/vnd.github.squirrel-girl-preview",
    "Accept: application/vnd.github.mockingbird-preview"
})
```

Actually, I've been doing some tests just now, and setting this headers works! 🎉 
```
@Headers({
    "Accept: application/vnd.github.v3.full+json",
    "Accept: application/vnd.github.squirrel-girl-preview",
    "Accept: application/vnd.github.mockingbird-preview"
})
```
You can try using this to solve the issue.

> Founded one trouble while developing. Timeline refreshes for three or more different issues when user clicks on only one issue to view details. For example click on issue 300 fires 299, 300, 301 issue details updating. IMHO that's a real reason of slow refreshing repository's issue info.

This is due to the issue view being a ViewPager (you can scroll left or right to change issues), so Android loads more issues for smooth scrolling.

**Comment by vkavalchuk** [comment]
Thanks a lot for your advice!

**Comment by jonan** [comment]
You're welcome.
Ping me when you're ready for merge and I review it again 😉

**Comment by vkavalchuk** [comment]
I'm ready to merge. Reactions for a cimmit's comment added. I'm not found separate wiev for a pull request's comment reaction. Issue #280 seems ready to be closed.

File: app/src/main/java/com/github/mobile/ui/commit/CommitFileListAdapter.java [hunk 1]
```diff
@@ -202,10 +203,12 @@ protected void update(final int position, final Object item, final int type) {
         case TYPE_COMMENT:
             CommitComment comment = (CommitComment) item;
             avatars.bind(imageView(1), comment.getUser());
-            setText(2, comment.getUser().getLogin());
+            setText(2, comment.getUser().login);
             setText(3, TimeUtils.getRelativeTime(comment.getUpdatedAt()));
+            setGone(4, !comment.getUpdatedAt().after(comment.getCreatedAt()));
```
Comment (vkavalchuk) [inline app/src/main/java/com/github/mobile/ui/commit/CommitFileListAdapter.java]: Little fix is here.

File: app/src/main/java/com/github/mobile/api/model/CommitComment.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.squareup.moshi.Json;
+
+public class CommitComment extends Comment{
```
Comment (jonan) [inline app/src/main/java/com/github/mobile/api/model/CommitComment.java]: Same as `Comment`

**Comment by vkavalchuk** [comment]
OK, I'll rewrite/refactor my changes into another branch/PR with 3 commits.

**Comment by vkavalchuk** [comment]
There is 4th change - refresh issue optimization. Should it be placed in separate commit/PR or not?

**Comment by vkavalchuk** [comment]
Refactored pull is here #307

**Pull request closed without merging**